### PR TITLE
musikcube: fix rpath for musikcube

### PR DESCRIPTION
**Pull request by afh**
###### Description of changes

I've filed [clangen/musikcube#538](https://github.com/clangen/musikcube/issues/538), since the latest version of musikcube aborts on Darwin when installed with nixpkgs due to an incorrect/unresolved `rpath`.
At the time of writing I'm uncertain whether this is an upstream issue with musikcube that was introduced with 0.98.0 or something in nix/nixpkgs has changed.

Putting up this draft PR for review to get a conversation going how and where this issue would be resolved best.

Any suggestions for improvement in this PR are welcome.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/audio/musikcube/default.nix [hunk 1]
```diff
@@ -76,6 +76,11 @@ stdenv.mkDerivation rec {
     "-DDISABLE_STRIP=true"
   ];
 
+  postFixup = if stdenv.hostPlatform.isDarwin then ''
+    install_name_tool -add_rpath $out/share/${pname} $out/share//${pname}/${pname}
```
Comment (aanderse) [inline pkgs/applications/audio/musikcube/default.nix:80]: Did you mean to have double slashes?

**Comment by aanderse** [comment]
Would you like to add yourself as a maintainer?

**Comment by aanderse** [comment]
> Putting up this draft PR for review to get a conversation going how and where this issue would be resolved best.

I'm sorry I have no knowledge of this platform so I can't comment.

File: pkgs/applications/audio/musikcube/default.nix (hunk 1 above)
Comment (afh) [reply pkgs/applications/audio/musikcube/default.nix]: I did not, thanks for pointing this out, @anderse.
I also seized the opportunity to improve this Darwin specific `postFixup` a bit.

**Comment by afh** [comment]
> Would you like to add yourself as a maintainer?

Oh? Thank you 😄 To help me decide what are the rights and duties of a nxipkg maintainer?

> I'm sorry I have no knowledge of this platform so I can't comment.

I'll see what the conversation on the [issue in the upstream repo](https://github.com/clangen/musikcube/issues/538) surfaces. It's good to know that at least a workaround, i.e. the changes in this PR, exists now.

**Comment by afh** [comment]
I've marked this PR ready for review @aanderse, as I believe the pragmatic path forward is to merge the changes proposed in this PR. After some input from the upstream maintainer and thinking about it a bit more the workaround here seems to solve the issue good enough for the time being.

If I find a better upstream fix for this I'll propose the changes upstream and here as necessary.

**Comment by afh** [comment]
Thanks for merging @aanderse 🙏🙂

**Pull request merged**